src/libcore/num/mod.rs

```diff
@@ -1529,12 +1529,8 @@ assert_eq!(", stringify!($SelfT), "::MIN.overflowing_neg(), (", stringify!($Self
 ```"),
             #[inline]
             #[stable(feature = "wrapping", since = "1.7.0")]
-            pub fn overflowing_neg(self) -> (Self, bool) {
-                if self == Self::min_value() {
-                    (Self::min_value(), true)
-                } else {
-                    (-self, false)
-                }
+            pub const fn overflowing_neg(self) -> (Self, bool) {
+                ((self ^ -1).wrapping_add(1), self == Self::min_value())
```

Comment:
`^ -1` is the same as `!`. With that insight both signed and unsigned versions can be made to look almost the same.

Reply:
Got it